src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -53,38 +56,64 @@
      */
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
-        Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
-        PreparedQuery userComments = datastore.prepare(query);
+        Entity currentDeed = getCurrentDeed();
+        Collection userComments = (Collection) currentDeed.getProperty("Comments");
+        Iterator<String> commentsIterator = userComments.iterator();
```

Comment:
you can avoid having to get the iterator and simply do [`comments.addAll()`](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#addAll-java.util.Collection-). So below you can replaced the while loop with `comments.addAll(userComments)`

Reply:
Cool! I will do that